c7n/commands.py

```diff
@@ -232,12 +236,32 @@ def validate(options):
                 )
             ))
         used_policy_names = used_policy_names.union(conf_policy_names)
+        source_locator = None
+        if fmt[0] == "y":
+            # For yaml files there is at least the expectation that the policy
+            # name is on a line by itself. With JSON, the file could be one big
+            # line. At this stage we are only attempting to find line number for
+            # policies in yaml files.
+            source_locator = SourceLocator(config_file)
         if not errors:
             null_config = Config.empty(dryrun=True, account_id='na', region='na')
             for p in data.get('policies', ()):
                 try:
                     policy = Policy(p, null_config, Bag())
                     policy.validate()
+                    # If the policy is invalid, there isn't much point checking
+                    # for deprecated usage as there is no guarantee as to the
+                    # state of the policy.
+                    if options.check_deprecations != "skip":
```

Comment:
would this work as  tri-valued Boolean logic (null, true, false) or a string variable constant ?

Reply:
I think these would be better as string constants rather than tri-valued Boolean logic, as the behaviour doesn't nicely map to true/false/null values. At least I couldn't think of a variable name where those values mapped to required behaviour.